sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -40,7 +42,16 @@
     private static final int MAX_BODY_LOG_SIZE = 1024 * 16;
     private static final String REDACTED_PLACEHOLDER = "REDACTED";
 
-    private final ClientLogger logger = new ClientLogger(HttpLoggingPolicy.class);
+    // Use a cache to retain the caller method ClientLogger.
+    //
+    // The same method may be called thousands or millions of times, so it is wasteful to create a new logger instance
+    // each time the method is called. Instead, retain the created ClientLogger until a certain number of unique method
+    // calls have been made and then clear the cache and rebuild it. Long term, this should be replaced with an LRU,
+    // or another type of cache, for better cache management.
```

Comment:
Should we go ahead and implement thread safe LRU cache? we already have couple of places where this could be useful.
It doesn't seem to be very complicated http://www.java2s.com/example/java/data-structure/concurrent-lru-cache.html .